src/traces/scatter/fillcolor_defaults.js

```diff
@@ -18,9 +27,25 @@ module.exports = function fillColorDefaults(traceIn, traceOut, defaultColor, coe
         }
     }
 
+    var averageGradientColor;
+    if(traceIn.fillgradient) {
+        // if a fillgradient is specified, we use the average gradient color
+        // to specifiy fillcolor after all other more specific candidates
+        // are considered, but before the global default color.
+        // fillcolor affects the background color of the hoverlabel in this case.
+        var gradientOrientation = coerce('fillgradient.type');
+        if(gradientOrientation !== 'none') {
+            coerce('fillgradient.start');
+            coerce('fillgradient.stop');
+            coerce('fillgradient.colorscale');
+            averageGradientColor = averageColors(traceOut.fillgradient.colorscale);
+        }
+    }
```

Comment:
```suggestion
    // if a fillgradient is specified, we use the average gradient color
    // to specifiy fillcolor after all other more specific candidates
    // are considered, but before the global default color.
    // fillcolor affects the background color of the hoverlabel in this case.
    var gradientOrientation = coerce('fillgradient.type');
    if(gradientOrientation !== 'none') {
        coerce('fillgradient.start');
        coerce('fillgradient.stop');
        coerce('fillgradient.colorscale');
        averageGradientColor = averageColors(traceOut.fillgradient.colorscale);
    }
```

Reply:
Let me add a commit fixing that.

Reply:
Please pick https://github.com/plotly/plotly.js/commit/a2400f16ac2645222aa83cbcb532dbe626ede1d3